src/module/actor/actor-sheet.js

```diff
@@ -533,8 +533,39 @@ export class OseActorSheet extends ActorSheet {
       item.sheet.render(true);
     });
 
-    html.find(".item-delete").click((event) => {
-      this._removeItemFromActor(event);
+    html.find(".item-delete").click(async (event) => {
+      const item = this._getItemFromActor(event);
+      const itemData = item?.system;
+
+      if (item.type === "container" && itemData.itemIds.length > 0) {
+        // Create dialog
+        let templateData = { types:["ok", "cancel"] },
+          dlg = await renderTemplate(
+            `${OSE.systemPath()}/templates/items/delete-container.html`,
+            templateData
+          );
+  
+        const confirmation = new Dialog({
```

Comment:
You should be able to use `Dialog.confirm` for this: https://foundryvtt.com/api/v10/classes/client.Dialog.html#confirm

Reply:
So much more elegant, thanks, I have now changed to using `Dialog.config` instead.